lib/jsdom/level2/html.js

```diff
@@ -2096,6 +2097,36 @@ define('HTMLFrameElement', {
         }
       }
     },
+
+    _detach: function () {
+      core.HTMLElement.prototype._detach.call(this);
+
+      var self = this;
+      var parent = this._ownerDocument.defaultView;
+
+      if (self.contentWindow) {
+        self.contentWindow.close();
+      }
+
+      // delete frame reference
+      if (self.getAttribute("name")) {
+        parent._frame(self.getAttribute("name"));
+      }
+
+      // delete index reference by moving all frames with a higher index one index down
+      for (var i = self._frameID; i < parent.length - 1; ++i) {
+        var next = parent.document.querySelectorAll("iframe,frame")[i];
```

Comment:
We need to rethink this, also with the frameIDs. What if frames are moved in the DOM, although this already would screw up the getters I think...